format.js

```diff
@@ -98,17 +99,24 @@ function makeEncryptionFormat() {
     const authorId = opts.keys.id
     const previousId = opts.previous
 
-    const validRecps = recps
+    const validPkRecps = recps
       .filter((recp) => typeof recp === 'string')
       .filter((recp) => recp === authorId || _isGroup(recp) || _isFeed(recp))
 
-    if (validRecps.length === 0) {
+    const validGroupkeyRecps = recps.filter(
+      (recp) =>
+        Buffer.isBuffer(recp.key) && recp.scheme === keySchemes.private_group
```

Comment:
```suggestion
        recp && Buffer.isBuffer(recp.key) && recp.scheme === keySchemes.private_group
```

I would add this just to make sure that `recp.key` doesn't explode in case someone put it `null` as a recp.

Other than that, looks safe to merge!